### PR TITLE
`ASAP-Alchemy`: Remove potential covalent ligands from CDD downloads

### PR DESCRIPTION
**Pull request by jthorton**
## Description
This PR extends the CDD downloader utility to flag potential covalent ligands to ensure they are not used as references in the RBFE calculations. 

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] Flag potential warheads and remove from the download

## Questions
- [ ] Are there any other ways we could flag the ligands? 

## Status
- [x] Ready to go

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Review by JenkeScheen — APPROVED** [review]
LGTM pending approval from MedChemists!

**Review by hmacdope — APPROVED** [review]
LGTM

**Pull request merged**
